File: wrappers/dotnet/indy-sdk-dotnet-test/SignusTests/SetKeyMetadataTest.cs
using Hyperledger.Indy.SignusApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.SignusTests
{
    [TestClass]
    public class SetKeyMetadataTest : IndyIntegrationTestWithSingleWallet
    {
        [TestMethod]
        public async Task TestSetKeyMetadataWorks()
        {
            await Signus.SetKeyMetadataAsync(wallet, VERKEY, METADATA);
        }

        [TestMethod]
        public async Task TestSetKeyMetadataWorksForReplace()
        {
            await Signus.SetKeyMetadataAsync(wallet, VERKEY, METADATA);
            var receivedMetadata = await Signus.GetKeyMetadataAsync(wallet, VERKEY);
            Assert.AreEqual(METADATA, receivedMetadata);

            var newMetadata = "updated metadata";
            await Signus.SetKeyMetadataAsync(wallet, VERKEY, newMetadata);
            var updatedMetadata = await Signus.GetKeyMetadataAsync(wallet, VERKEY);

            Assert.AreEqual(newMetadata, updatedMetadata);
        }

        [TestMethod]
        public async Task TestSetKeyMetadataWorksForEmptyString()
        {
            await Signus.SetKeyMetadataAsync(wallet, VERKEY, string.Empty);
        }

    }
}

File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/signus/SetKeyMetadataTest.java
package org.hyperledger.indy.sdk.signus;

import org.hyperledger.indy.sdk.IndyIntegrationTestWithSingleWallet;
import org.junit.Test;

import static org.junit.Assert.assertEquals;


public class SetKeyMetadataTest extends IndyIntegrationTestWithSingleWallet {

	@Test
	public void testSetKeyMetadataWorks() throws Exception {
		Signus.setKeyMetadata(wallet, VERKEY, METADATA).get();
	}

	@Test
	public void testSetKeyMetadataWorksForReplace() throws Exception {
		Signus.setKeyMetadata(wallet, VERKEY, METADATA).get();
		String receivedMetadata = Signus.getKeyMetadata(wallet, VERKEY).get();
		assertEquals(METADATA, receivedMetadata);

		String newMetadata = "updated metadata";
		Signus.setKeyMetadata(wallet, VERKEY, newMetadata).get();
		String updatedMetadata = Signus.getKeyMetadata(wallet, VERKEY).get();
		assertEquals(newMetadata, updatedMetadata);
	}

	@Test
	public void testSetKeyMetadataWorksForEmptyString() throws Exception {
		Signus.setKeyMetadata(wallet, VERKEY, "").get();
	}
}